From 2f86e91a4f390fb6ebb0b4dae6574a4b4038dd60 Mon Sep 17 00:00:00 2001
From: Ding Wei <leo.ding@rock-chips.com>
Date: Wed, 8 Apr 2020 17:32:06 +0800
Subject: [PATCH] video: rockchip: mpp: fix issue for rkvenc request check

It is not necessary to check offset for MPP_CMD_SET_REG_ADDR_OFFSET,
and user space can not ensure it.

Change-Id: Iac3a2fa3ce8be8d9afd7294e7c3ec4bc6df4a2e3
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 drivers/video/rockchip/mpp/mpp_rkvenc.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/video/rockchip/mpp/mpp_rkvenc.c b/drivers/video/rockchip/mpp/mpp_rkvenc.c
index e49de9d4c527..b932e63b3234 100644
--- a/drivers/video/rockchip/mpp/mpp_rkvenc.c
+++ b/drivers/video/rockchip/mpp/mpp_rkvenc.c
@@ -267,9 +267,8 @@ static int rkvenc_extract_task_msg(struct rkvenc_task *task,
 			       req, sizeof(*req));
 		} break;
 		case MPP_CMD_SET_REG_ADDR_OFFSET: {
-			int off = off_inf->cnt * sizeof(off_inf->elem[0]);
-
-			ret = mpp_check_req(req, off, sizeof(off_inf->elem),
+			ret = mpp_check_req(req, req->offset,
+					    sizeof(off_inf->elem),
 					    0, sizeof(off_inf->elem));
 			if (ret)
 				return ret;
-- 
2.35.3

